### PR TITLE
e2e/workloads: refactor and simplify

### PR DESCRIPTION
**Pull request by jbpratt**
the workloads tests are simple enough to be a table test

Signed-off-by: Brady Pratt <bpratt@redhat.com>

closes [SDE-2420](https://issues.redhat.com//browse/SDE-2420)

### CODE REVIEW

**Comment by ryankwilliams** [comment]
For [SDCICD-882](https://issues.redhat.com/browse/SDCICD-882)

File: pkg/e2e/workloads/workloads.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	"sigs.k8s.io/e2e-framework/klient/decoder"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+)
+
+func init() {
+	alert.RegisterGinkgoAlert("Workloads", "SD-SREP", "@sd-qe", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
+}
+
+var _ = ginkgo.Describe("Workloads", ginkgo.Ordered, label.HyperShift, label.E2E, func() {
```
Comment (ryankwilliams) [inline pkg/e2e/workloads/workloads.go:27]:
```suggestion
var _ = ginkgo.Describe("[Suite: e2e] Workloads", ginkgo.Ordered, label.HyperShift, label.E2E, func() {
```

Just adding back the suite string, so these tests get picked as part of e2e run. Once we fully switch the configs over to label filters, we can remove these strings.

Comment (jbpratt) [reply pkg/e2e/workloads/workloads.go]: updated!

**Comment by ryankwilliams** [comment]
Verified the change locally (see junit results below):

1. OSD

```xml
          <testcase name="[Suite: e2e] Workloads can be created, used, and deleted Redmine" classname="OSD e2e suite" status="passed" time="20.919342953">
              <system-err>&gt; Enter [BeforeAll] [Suite: e2e] Workloads - /home/rywillia/osd/osde2e/pkg/e2e/workloads/workloads.go:32 @ 01/11/23 13:31:51.402&#xA;&lt; Exit [BeforeAll] [Suite: e2e] Workloads - /home/rywillia/osd/osde2e/pkg/e2e/workloads/workloads.go:32 @ 01/11/23 13:31:51.441 (39ms)&#xA;&gt; Enter [It] Redmine - /home/rywillia/osd/osde2e/pkg/e2e/workloads/workloads.go:56 @ 01/11/23 13:31:51.441&#xA;&lt; Exit [It] Redmine - /home/rywillia/osd/osde2e/pkg/e2e/workloads/workloads.go:56 @ 01/11/23 13:32:12.321 (20.88s)&#xA;</system-err>
          </testcase>
          <testcase name="[Suite: e2e] Workloads can be created, used, and deleted Guestbook" classname="OSD e2e suite" status="passed" time="15.824999159">
              <system-err>&gt; Enter [It] Guestbook - /home/rywillia/osd/osde2e/pkg/e2e/workloads/workloads.go:57 @ 01/11/23 13:32:12.322&#xA;&lt; Exit [It] Guestbook - /home/rywillia/osd/osde2e/pkg/e2e/workloads/workloads.go:57 @ 01/11/23 13:32:28.147 (15.825s)&#xA;</system-err>
          </testcase>
```


2. HyperShift

```xml
          <testcase name="[Suite: e2e] Workloads can be created, used, and deleted Redmine" classname="OSD e2e suite" status="passed" time="30.950900462">
              <system-err>&gt; Enter [BeforeAll] [Suite: e2e] Workloads - /home/rywillia/osd/osde2e/pkg/e2e/workloads/workloads.go:32 @ 01/11/23 13:35:44.201&#xA;&lt; Exit [BeforeAll] [Suite: e2e] Workloads - /home/rywillia/osd/osde2e/pkg/e2e/workloads/workloads.go:32 @ 01/11/23 13:35:44.245 (44ms)&#xA;&gt; Enter [It] Redmine - /home/rywillia/osd/osde2e/pkg/e2e/workloads/workloads.go:56 @ 01/11/23 13:35:44.245&#xA;&lt; Exit [It] Redmine - /home/rywillia/osd/osde2e/pkg/e2e/workloads/workloads.go:56 @ 01/11/23 13:36:15.152 (30.907s)&#xA;</system-err>
          </testcase>
          <testcase name="[Suite: e2e] Workloads can be created, used, and deleted Guestbook" classname="OSD e2e suite" status="passed" time="5.7908758559999995">
              <system-err>&gt; Enter [It] Guestbook - /home/rywillia/osd/osde2e/pkg/e2e/workloads/workloads.go:57 @ 01/11/23 13:36:15.152&#xA;&lt; Exit [It] Guestbook - /home/rywillia/osd/osde2e/pkg/e2e/workloads/workloads.go:57 @ 01/11/23 13:36:20.943 (5.791s)&#xA;</system-err>
          </testcase>
```

**Pull request merged**